### PR TITLE
Check preTags and postTags params for empty values

### LINKED ISSUE #69009: `pre_tags` and `post_tags` params not validated for empty values
`pre_tags` and `post_tags` query parameters for highlighting are not validated of empty values.
As a result if a query sets them to an empty array we get an ArrayOutOfBoundsException:

```
{
  "query" : {
    "match" : {
      "field1" : "fox"}
    }, 
    "highlight" : {
      "type" : "plain",
      "fields" : {
        "field1" : {}
      },
      "pre_tags": ["<tag1>"],
      "post_tags" : []
    }
}
```

results in:

```
...
"caused_by" : {
           "type" : "array_index_out_of_bounds_exception",
           "reason" : "Index 0 out of bounds for length 0",
           "caused_by" : {
             "type" : "array_index_out_of_bounds_exception",
             "reason" : "Index 0 out of bounds for length 0",
             "stack_trace" : "java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
    at org.elasticsearch.search.fetch.subphase.highlight.PlainHighlighter.highlight(PlainHighlighter.java:79)
    at org.elasticsearch.search.fetch.subphase.highlight.HighlightPhase$1.process(HighlightPhase.java:64)
    at org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:146)
    at org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:461)
    at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:437)
    at org.elasticsearch.search.SearchService$2.lambda$onResponse$0(SearchService.java:395)
    at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:47)
    at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:62)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservin gAbstractRunnable.doRun(ThreadContext.java:728)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
    at java.base/java.lang.Thread.run(Thread.java:832)

...

### PR DESCRIPTION
**Pull request by white-cc**
Check `pre_tags` and `post_tags` params for empty values.

If empty throw `IllegalArgumentException`.

Closes #69009

### CODE REVIEW

**Review by mayya-sharipova — CHANGES_REQUESTED** [review]
@white-cc Thanks for your contributions. This should be fixed for all highlighters, not only for  the`plain` highlighter.

I suggest to add the check in [AbstractHighlighterBuilder](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/AbstractHighlighterBuilder.java#L643) where there is already another check. 

Also please add rest tests into `rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.highlight` for all types of highlighters. Here is the guidance about [YAML rest tests](https://github.com/elastic/elasticsearch/blob/main/TESTING.asciidoc#testing-the-rest-layer).

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.highlight/issue69009.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+setup:
```
Comment (mayya-sharipova) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.highlight/issue69009.yml:1]: Let's add this setup section which means to skip the test for nodes with version < 8.14

setup:
  - skip:
      version: ' - 8.13.99'
      reason: 'check of preTags and postTags params for empty values was added in   8.14'

**Comment by mayya-sharipova** [comment]
@elasticsearchmachine test this please

File: docs/changelog/106396.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+pr: 106396
+summary: "fix check preTags and postTags with empty argument"
+area: search
```
Comment (mayya-sharipova) [inline docs/changelog/106396.yaml:3]: area: Highlighting

**Comment by mayya-sharipova** [comment]
@elasticsearchmachine Test this please

**Comment by mayya-sharipova** [comment]
@white-cc Thanks for your work so far, looks great. 

Are you able to access CI build failures to check and fix tests that are failing:

```

REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.index.query.InnerHitBuilderTests.testFromAndToXContent" -Dtests.seed=99449CFAF2B54821 -Dtests.locale=sr-Latn-BA -Dtests.timezone=America/Indiana/Indianapolis -Druntime.java=21
--
  |  
  | org.elasticsearch.index.query.InnerHitBuilderTests > testFromAndToXContent FAILED
  | org.elasticsearch.xcontent.XContentParseException: [1:459] [inner_hits] failed to parse field [highlight]
  | at __randomizedtesting.SeedInfo.seed([99449CFAF2B54821:DA77E6EBAE3E418C]:0)
  | at app//org.elasticsearch.xcontent.ObjectParser.throwFailedToParse(ObjectParser.java:616)
  | at app//org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:611)
  | at app//org.elasticsearch.xcontent.ObjectParser.parseSub(ObjectParser.java:629)
  | at app//org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:315)
  | at app//org.elasticsearch.index.query.InnerHitBuilder.fromXContent(InnerHitBuilder.java:597)
  | at app//org.elasticsearch.index.query.InnerHitBuilderTests.testFromAndToXContent(InnerHitBuilderTests.java:119)
  |  
  | Caused by:
  | org.elasticsearch.xcontent.XContentParseException: [1:459] [highlight] failed to parse field [fields]
  | at app//org.elasticsearch.xcontent.ObjectParser.throwFailedToParse(ObjectParser.java:616)
  | at app//org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:611)
  | at app//org.elasticsearch.xcontent.ObjectParser.parseSub(ObjectParser.java:629)
  | at app//org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:315)
  | at app//org.elasticsearch.search.fetch.subphase.highlight.AbstractHighlighterBuilder.lambda$setupParser$4(AbstractHighlighterBuilder.java:642)
  | at app//org.elasticsearch.search.fetch.subphase.highlight.HighlightBuilder.fromXContent(HighlightBuilder.java:266)
  | at app//org.elasticsearch.index.query.InnerHitBuilder.lambda$static$6(InnerHitBuilder.java:107)
  | at app//org.elasticsearch.xcontent.ObjectParser.lambda$declareField$10(ObjectParser.java:431)
  | at app//org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:609)
  | ... 4 more
  |  
  | Caused by:
  | org.elasticsearch.xcontent.XContentParseException: [1:459] [fields] failed to parse field [0_sYNHwBMlvn]
  | at app//org.elasticsearch.xcontent.ObjectParser.rethrowFieldParseFailure(ObjectParser.java:554)
  | at app//org.elasticsearch.xcontent.ObjectParser.lambda$declareNamedObjects$14(ObjectParser.java:494)
  | at app//org.elasticsearch.xcontent.ObjectParser.lambda$declareNamedObjects$15(ObjectParser.java:505)
  | at app//org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:609)
  | ... 11 more
  |  
  | Caused by:
  | org.elasticsearch.common.ParsingException: pre_tags or post_tags must not be empty


````

**Comment by white-cc** [comment]
> @white-cc Thanks for your work so far, looks great.
> 
> Are you able to access CI build failures to check and fix tests that are failing:
> 
> ```
> 
> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.index.query.InnerHitBuilderTests.testFromAndToXContent" -Dtests.seed=99449CFAF2B54821 -Dtests.locale=sr-Latn-BA -Dtests.timezone=America/Indiana/Indianapolis -Druntime.java=21
> --
>   |  
>   | org.elasticsearch.index.query.InnerHitBuilderTests > testFromAndToXContent FAILED
>   | org.elasticsearch.xcontent.XContentParseException: [1:459] [inner_hits] failed to parse field [highlight]
>   | at __randomizedtesting.SeedInfo.seed([99449CFAF2B54821:DA77E6EBAE3E418C]:0)
>   | at app//org.elasticsearch.xcontent.ObjectParser.throwFailedToParse(ObjectParser.java:616)
>   | at app//org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:611)
>   | at app//org.elasticsearch.xcontent.ObjectParser.parseSub(ObjectParser.java:629)
>   | at app//org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:315)
>   | at app//org.elasticsearch.index.query.InnerHitBuilder.fromXContent(InnerHitBuilder.java:597)
>   | at app//org.elasticsearch.index.query.InnerHitBuilderTests.testFromAndToXContent(InnerHitBuilderTests.java:119)
>   |  
>   | Caused by:
>   | org.elasticsearch.xcontent.XContentParseException: [1:459] [highlight] failed to parse field [fields]
>   | at app//org.elasticsearch.xcontent.ObjectParser.throwFailedToParse(ObjectParser.java:616)
>   | at app//org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:611)
>   | at app//org.elasticsearch.xcontent.ObjectParser.parseSub(ObjectParser.java:629)
>   | at app//org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:315)
>   | at app//org.elasticsearch.search.fetch.subphase.highlight.AbstractHighlighterBuilder.lambda$setupParser$4(AbstractHighlighterBuilder.java:642)
>   | at app//org.elasticsearch.search.fetch.subphase.highlight.HighlightBuilder.fromXContent(HighlightBuilder.java:266)
>   | at app//org.elasticsearch.index.query.InnerHitBuilder.lambda$static$6(InnerHitBuilder.java:107)
>   | at app//org.elasticsearch.xcontent.ObjectParser.lambda$declareField$10(ObjectParser.java:431)
>   | at app//org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:609)
>   | ... 4 more
>   |  
>   | Caused by:
>   | org.elasticsearch.xcontent.XContentParseException: [1:459] [fields] failed to parse field [0_sYNHwBMlvn]
>   | at app//org.elasticsearch.xcontent.ObjectParser.rethrowFieldParseFailure(ObjectParser.java:554)
>   | at app//org.elasticsearch.xcontent.ObjectParser.lambda$declareNamedObjects$14(ObjectParser.java:494)
>   | at app//org.elasticsearch.xcontent.ObjectParser.lambda$declareNamedObjects$15(ObjectParser.java:505)
>   | at app//org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:609)
>   | ... 11 more
>   |  
>   | Caused by:
>   | org.elasticsearch.common.ParsingException: pre_tags or post_tags must not be empty
> ```

Yes, I tested it locally and found the exception. pre_tags is empty in this case, but it may case "Index 0 out of bounds for length 0" in my test. I will try to modify code to avoid  pre_tags empty in this uint test.

image

**Comment by mayya-sharipova** [comment]
@elasticsearchmachine test this please

**Review by mayya-sharipova — APPROVED** [review]
@white-cc Thanks for your contribution! This PR looks good to me. Merging for 8.14 release.

**Pull request merged**